app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+  before_save :set_slug
```

Comment:
this will recalculate the slug every time you persist a user change. Maybe you want `before_create`?